jiant/proj/simple/runscript.py

```diff
@@ -234,148 +243,19 @@ def run_simple(args: RunConfiguration):
-    --epochs {args.num_train_epochs} \\
-    --num_gpus {torch.cuda.device_count()}
-""".strip()
-    if args.train_examples_cap:
-        s += f" \\\n    --train_examples_cap {args.train_examples_cap}"
-    print(s.strip())
-
-    print("\n# === Step 5: Train/Eval! === #")
-    if args.model_weights_path:
-        model_load_mode = "partial"
-        model_weights_path = args.model_weights_path
-    else:
-        # From Transformers
-        if any(task_name.startswith("mlm_") for task_name in full_task_name_list):
-            model_load_mode = "from_transformers_with_mlm"
-        else:
-            model_load_mode = "from_transformers"
-        model_weights_path = os.path.join(
-            model_cache_path, args.model_type, "model", f"{args.model_type}.p"
-        )
-    s = f"""
-python jiant/proj/main/runscript.py \\
-    run \\
-    --jiant_task_container_config_path \
-{os.path.join(args.exp_dir, "run_configs", f"{args.run_name}_config.json")} \\
-    --output_dir {os.path.join(args.exp_dir, "runs", args.run_name)} \\
-    --model_type {args.model_type} \\
-    --model_path {model_weights_path} \\
-    --model_config_path \
-    {os.path.join(model_cache_path, args.model_type, "model", f"{args.model_type}.json")} \\
-    --model_tokenizer_path {os.path.join(model_cache_path, args.model_type, "tokenizer")} \\
-    --model_load_mode {model_load_mode}
-""".strip()
-    if args.train_tasks:
-        s += " \\\n    --do_train"
-    if args.val_tasks:
-        s += " \\\n    --do_val"
-    covered_attrs = [
-        "jiant_task_container_config_path",
-        "output_dir",
-        "model_type",
-        "model_path",
-        "model_config_path",
-        "model_tokenizer_path",
-        "model_load_mode",
-    ]
-    for attr in runscript.RunConfiguration.__attrs_attrs__:
-        if attr.name in covered_attrs:
-            continue
-        if not hasattr(args, attr.name):
-            continue
-        args_attr = getattr(args, attr.name)
-        if attr.default == args_attr:
-            continue
-        if attr.default is None and args_attr is None:
-            continue
-        if (
-            "argparse_kwargs" in attr.metadata
-            and "action" in attr.metadata["argparse_kwargs"]
-            and attr.metadata["argparse_kwargs"]["action"] == "store_true"
-        ):
-            s += f" \\\n    --{attr.name}"
-        else:
-            s += f" \\\n    --{attr.name} {args_attr}"
-    print(s.strip())
+    runscript.run_loop(args=run_args, checkpoint=checkpoint)
+    py_io.write_file(args.to_json(), os.path.join(run_output_dir, "simple_run_config.json"))
 
 
 def main():
     mode, cl_args = zconf.get_mode_and_cl_args()
     args = RunConfiguration.default_run_cli(cl_args=cl_args)
     if mode == "run":
-        run_simple(args)
-    elif mode == "dry_run":
-        dry_run(args)
+        run_simple(args, with_continue=False)
+    if mode == "run_with_continue":
+        run_simple(args, with_continue=True)
```

Comment:
Do you have the bandwidth to add a CI test to use this functionality? I can add it after as well.

Reply:
Let's do it after? Potentially we can alter the way it's called to make it easier to isolate+test.